datadog_checks_base/datadog_checks/base/checks/win/wmi/sampler.py

```diff
@@ -261,7 +262,9 @@ def sample(self):
         """
         self._sampling = True
         self._runSampleEvent.set()
-        self._sampleCompleteEvent.wait()
+        while not self._sampleCompleteEvent.wait(timeout=float(self._timeout_duration)):
```

Comment:
Just for the record of that PR. This function used to return None all the time. This was change with version 2.7 of python, now the function returns True if the event is set, False if the wait timed out.

Because the agent only uses Python 2.7 and 3.8, this change is good.